src/taskEvents/script.js

```diff
@@ -0,0 +1,283 @@
+    const element = createElement({
+      type: 'div',
+      attributes: { class: 'roles-div-item' },
+      innerText: role,
+    });
+    const removeBtn = createElement({
+      type: 'button',
+      attributes: { class: 'remove-btn', id: index },
+      innerText: 'x',
+    });
+    removeBtn.addEventListener('click', removeSkill);
+    element.appendChild(removeBtn);
+    rolesDiv.append(element);
+  });
+  const addBtn = createElement({
+    type: 'button',
+    attributes: { class: 'add-btn' },
+    innerText: '+',
+  });
+  addBtn.addEventListener('click', openAddSkillModal);
+  rolesDiv.appendChild(addBtn);
+  return rolesDiv;
+}
+
+function createUserActivityBtn(username) {
+  const activityBtnDiv = createElement({
+    type: 'div',
+    attributes: { class: 'activity-btn-div' },
+  });
+  const activityBtn = createElement({
+    type: 'button',
+    attributes: { class: 'activity-btn' },
+    innerText: 'show user activity',
+  });
+  activityBtn.addEventListener('click', () => openUserActivityModal(username));
+  activityBtnDiv.appendChild(activityBtn);
+  return activityBtnDiv;
+}
+
+function closeModal() {
+  overlay.style.display = 'none';
+  const errorDiv = document.querySelector('.error-div');
+  if (errorDiv) {
+    errorDiv.remove();
+    return;
+  }
+  document.querySelector('.roles-div').remove();
+  document.querySelector('.activity-btn-div').remove();
+  document.querySelector('.username').remove();
+  document.querySelector('.skill-title').remove();
+  document.querySelector('.user-img').remove();
+}
+
+function closeGenericModal() {
+  document.getElementById('generic-modal').style.visibility = 'collapse';
+  const containerDiv = document.getElementById('container-div');
+  containerDiv.innerHTML = '';
+}
+
+function removeSkill(e) {
+  // this is not how it's going to be done, when we have the userSkills this function is going to change
+  e.target.parentElement.remove();
+}
+
+function openAddSkillModal() {
+  const skills = ['frontend', 'backend', 'System Design']; // this is temporary data that will be removed once we have userSkill collection in our DB
+  const level = [1, 2, 3, 4, 5];
+  // a submit button which will make a request to the backend and save the skill in userSkills collection
+  const containerDiv = document.getElementById('container-div');
+  document.getElementById('generic-modal').style.visibility = 'visible';
+  const mainTitle = createElement({
+    type: 'h1',
+    attributes: { class: 'main-title' },
+    innerText: 'Add Skills',
+  });
+  const skillCategoryDiv = createElement({
+    type: 'div',
+    attributes: { class: 'skill-category-div' },
+  });
+  const skillCategoryLabel = createElement({
+    type: 'p',
+    attributes: { class: 'skill-category-label' },
+    innerText: 'Choose a skill to add',
+  });
+  const skillCategorySelect = createElement({
+    type: 'select',
+    attributes: { class: 'skill-category-select' },
+  });
+
+  for (let i = 0; i < skills.length; i++) {
```

Comment:
any reason to keep it **let** not **const**

Reply:
It will give an errro `Assignment to constant variable.`

Reply:
Got it